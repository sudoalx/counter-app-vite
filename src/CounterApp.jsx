import { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  /*
    Tarea: Manejar butones de resta y reset con hooks
  */

  const handleSubstract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <div className="counter-display">
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
      </div>
      <div className="button-controls">
        <button aria-label="btn-minus" onClick={() => handleSubstract()}>
          -1
        </button>
        <button aria-label="btn-reset" onClick={() => handleReset()}>
          Reset
        </button>
        <button aria-label="btn-plus" onClick={() => handleAdd()}>
          +1
        </button>
      </div>
    </>
  );
};

CounterApp.defaultProps = {
  value: 0,
};

CounterApp.defaultProps = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
