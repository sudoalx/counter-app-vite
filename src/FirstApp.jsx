import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name, age }) => {
  return (
    <>
      {/* Se usa el atributo data-testid para hacer tests por medio del test id en Jest */}
      <h1 data-testid="test-title">{title}</h1>
      <h3>{subTitle}</h3>
      <h3>{subTitle}</h3>
      <p>
        Hello {name}, {age} years old
      </p>
      <p>Year of birth: {age ? new Date().getFullYear() - age : age}</p>
    </>
  );
};

// Las definiciones de default props y props son definidas después del código a ejecutar
// Definir default props. Default props se definen antes de prop types
FirstApp.defaultProps = {
  title: "No title provided",
  subTitle: "No subtitle provided",
  name: "No name provided",
  age: 0,
};

// Los proptypes se definen de la siguiente manera:
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired, // Si se pasa un numero como String, automaticamente se convierte a number
};
