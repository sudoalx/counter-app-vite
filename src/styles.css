html,
body {
  background-color: #21232a;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  margin: 0;
}

h1 {
  font-size: 2em;
}

h2 {
  font-size: 1.5em;
}

h3 {
  font-size: 1.2em;
}

h4 {
  font-size: 1.1em;
}

h5 {
  font-size: 1em;
}

h6 {
  font-size: 1em;
}

.counter-display {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.button-controls {
  display: flex;
  align-items: center;
  justify-content: center;
}

button {
  padding: 15px;
  margin: 0.3rem;
  font-size: 1.3rem;
  border-radius: 1rem;
  border: 1px solid #2d3f83;
  background-color: #2d3f83;
  cursor: pointer;
  color: #ececec;
  transition: background-color 0.3s, color 0.3s;
}

button:hover {
  background-color: #1a2a55;
  color: #ffffff;
}
