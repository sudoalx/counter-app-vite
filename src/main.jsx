import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";
import CounterApp from "./CounterApp";

import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* Pasar props de title, subtitle y age al componente FirstApp */}
    {/* <FirstApp title="Home Screen" /> */}
    <CounterApp value={0} />
  </React.StrictMode>
);
