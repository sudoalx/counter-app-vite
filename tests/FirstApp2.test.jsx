import { render, screen } from "@testing-library/react"; // import screen
import { FirstApp } from "../src/FirstApp";
describe("Pruebas en FirstApp component", () => {
  const title = "Home Screen";
  const subTitle = "Subtitle";
  test("should match the snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test('should display the message "Home Screen"', () => {
    render(<FirstApp title={title} />); // Se puede usar reder() para renderizar el componente
    expect(screen.getByText(title)).toBeTruthy(); // Se puede usar screen para buscar elementos en el DOM
  });
  test("should show the title in an h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });
  test("should show the subtitle sent by props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
