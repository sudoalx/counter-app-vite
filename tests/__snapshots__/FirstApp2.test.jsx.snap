// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp component should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Home Screen
  </h1>
  <h3>
    No subtitle provided
  </h3>
  <h3>
    No subtitle provided
  </h3>
  <p>
    Hello 
    No name provided
    , 
    0
     years old
  </p>
  <p>
    Year of birth: 
    0
  </p>
</div>
`;
